### DISCUSSION TITLE
how to export mxnet model to json and pamas format ?

### DISCUSSION THREAD

**Discussion by Gpwner** [dmlc/gluon-nlp#658] [Q&A] [answered]
when I am using the code for bert in https://github.com/dmlc/gluon-nlp/tree/master/scripts/bert,
I wonder whether I can export mxnet model to [*.json +*.pamas] where the json file save model architecture and the pamas save parameters of the model.
Because I want to convert the finetune model of mxnet to pb format of tensorflow.
Thanks if anyone can help.

**Comment by szha** [comment, marked as answer]
@Gpwner currently it requires a small amount of manual process, where you may need to use the utility classes to construct a HybridBlock-based BERT model. The guide can be found here: http://gluon-nlp.mxnet.io/model_zoo/bert/index.html#example-usage-of-exporting-hybridizable-bert

We know this is not perfect, as it would be better to have an export API to automate this for all model classes in BERT model zoo. I created an issue to track this here: #659

**Comment by Gpwner** [comment]
> @Gpwner currently it requires a small amount of manual process, where you may need to use the utility classes to construct a HybridBlock-based BERT model. The guide can be found here: http://gluon-nlp.mxnet.io/model_zoo/bert/index.html#example-usage-of-exporting-hybridizable-bert
> 
> We know this is not perfect, as it would be better to have an export API to automate this for all model classes in BERT model zoo. I created an issue to track this here: #659

Thanks I will have a try.

**Comment by szha** [comment]
@Gpwner we now have a single script for exporting BERT for inference thanks to @eric-haibin-lin (#705). Feel free to give it a try and let us know how you like it :)
